modules/system/defaults-write.nix

```diff
@@ -10,6 +10,10 @@ let
 
   defaultsToList = domain: attrs: mapAttrsToList (writeDefault domain) (filterAttrs (n: v: v != null) attrs);
 
+  # Filter out options to not pass through
+  # dock has alias options that we need to ignore
+  dockFiltered = (builtins.removeAttrs cfg.dock ["expose-group-by-app"]);
```

Comment:
Without this, does that mean when you run it locally it sets both values?

Reply:
Can you use `mkRemovedOption` instead then?

Reply:
I tried that and it functioned the same, except with an assertion instead of a warning.

Reply:
From the code you sent, it looks like there wouldn't be a value set for the old option name when you use `mkRemovedOptionModule`

Reply:
There's a couple different open PRs to try and implement a better solution for trying to handle passing metadata around with options so eventually we don't have to do stuff like this https://github.com/NixOS/nixpkgs/pull/341199

Reply:
> From the code you sent, it looks like there wouldn't be a value set for the old option name when you use `mkRemovedOptionModule`

I would have assumed, too. But, it throws the error that we're trying to use a removed option.

Reply:
How about refactoring the code to iterate through `options` and filter out hidden fields, so we don't have to maintain a separate list?

Reply:
That's the original approach I was going to go through, but was convinced, during brainstorming with someone else, to go this route since a hidden value isn't specific to an alias rename. I can go that route if that's what you'd prefer in this repo.

Reply:
I'm curious what other situations would construct `hidden` values that we wouldn't want filtered out

Reply:
Sometimes you have public values that a user can use to configure something and you use a hidden option to construct the final value.

Reply:
That should be fine as I don't think we would use any of those for `system.defaults`

I'm not sure if you're interested, one wishlist item I have for refactoring the `defaults` code is to properly integrate `CustomUserPreferences` and `CustomSystemPreferences`, by moving all the options to be under `system.defaults.system.<...>` and `system.defaults.user.<...>`, making all the existing options be live inside there, using the freeform settings type as well as submodules to allow each group to define its domain

Something like:

```nix
{
  options.system.defaults.system.alf.globalstate = mkOption {
   ...
  };

  config.system.defaults.system.alf.domain = "/Library/Preferences/com.apple.alf";
}
```

Reply:
I'm fine with leaving this as is as the list is currently just this one renamed option

Reply:
> I'm fine with leaving this as is as the list is currently just this one renamed option

Sounds good, I started a branch for refactoring the defaults-write stuff for merging and using freeform type support. But, I am still working on some vimPlugins nixpkgs stuff atm that's keeping me mentally from being able to focus on other work atm.